src/components/WalletModal/index.tsx

```diff
@@ -159,6 +200,25 @@ export default function WalletModal({
   const walletModalOpen = useModalIsOpen(ApplicationModal.WALLET)
   const toggleWalletModal = useToggleWalletModal()
 
+  const allTokens = useAllTokens()
+  const [tokenBalances, tokenBalancesIsLoading] = useAllTokenBalances()
+  const sortedTokens: Token[] = useMemo(
+    () => (!tokenBalancesIsLoading ? Object.values(allTokens).sort(tokenComparator.bind(null, tokenBalances)) : []),
+    [tokenBalances, allTokens, tokenBalancesIsLoading]
+  )
+  const native = useNativeCurrency()
+
+  const sortedTokensWithETH: Currency[] = useMemo(
+    () =>
+      // Always bump the native token to the top of the list.
+      native ? [native, ...sortedTokens.filter((t) => !t.equals(native))] : sortedTokens,
```

Comment:
i think this logic is also used above? https://github.com/Uniswap/interface/pull/4206/files#diff-a3bebdfee25c1ff21ece35285ea88c59da706d32fa738ae2a1941d2e762c7039R129 perhaps we should make a funciton for this

Reply:
its just one line thats the same ([native, ...sortedTokens.filter((t) => !t.equals(native))]) seems a little too small to extract imo.